backend/tests/v5/e2e/routes/sso/aad/aad.test.js

```diff
@@ -254,6 +254,181 @@ const signupPost = () => {
 	});
 };
 
+const testLink = () => {
+	describe('Link', () => {
+		const redirectUri = generateRandomURL();
+
+		test('should fail without a valid session', async () => {
+			await agent.get(`/v5/sso/aad/link?redirectUri=${redirectUri}&key=${testUser.apiKey}`)
+				.expect(templates.notAuthorized.status);
+		});
+
+		describe('With valid authentication', () => {
+			beforeAll(async () => {
+				await testSession.post('/v5/login/').send({ user: testUser.user, password: testUser.password });
+			});
+			afterAll(async () => {
+				await testSession.post('/v5/logout/');
+			});
+
+			test('should fail without a valid state', async () => {
+				const state = generateRandomString();
+				const res = await testSession.get(`/v5/sso/aad/link-post?state=${state}`)
+					.expect(templates.invalidArguments.status);
+				expect(res.body.code).toEqual(templates.invalidArguments.code);
+			});
+
+			test('should fail if redirectUri is not provided', async () => {
+				await testSession.get('/v5/sso/aad/link')
+					.expect(templates.invalidArguments.status);
+			});
+
+			test('should redirect the user to Microsoft authentication page', async () => {
+				const res = await testSession.get(`/v5/sso/aad/link?redirectUri=${redirectUri}`)
+					.expect(302);
+				const resUri = new URL(res.headers.location);
+				expect(resUri.hostname).toEqual('login.microsoftonline.com');
+				expect(resUri.pathname).toEqual('/common/oauth2/v2.0/authorize');
+				expect(resUri.searchParams.get('redirect_uri')).toEqual(linkRedirectUri);
+				expect(resUri.searchParams.has('client_id')).toEqual(true);
```

Comment:
less clutter if you just put searchParams in a variable since you have so many lines to test